File: codigo/matlab/trapecio.m
function I = trapecio(f, a, b, n)
% TRAPECIO  Regla del trapecio compuesta. Chapra cap. 21.
%   Uso:  I = trapecio(@sin, 0, pi, 1000);
    if nargin < 4, n = 100; end
    h = (b - a)/n;
    x = a:h:b;
    y = f(x);
    I = h*(0.5*y(1) + sum(y(2:end-1)) + 0.5*y(end));
end

File: codigo/python/trapecio.py
"""trapecio.py -- Integracion por la regla del trapecio compuesta. Chapra cap. 21."""


def trapecio(f, a, b, n=100):
    h = (b - a) / n
    s = 0.5 * (f(a) + f(b))
    for i in range(1, n):
        s += f(a + i * h)
    return s * h


if __name__ == "__main__":
    import math
    I = trapecio(math.sin, 0, math.pi, 1000)
    print(f"Integral de sin en [0, pi] = {I:.8f}   (exacto = 2)")
